apps/web/components/Templates/TemplateCard.tsx

```diff
@@ -0,0 +1,56 @@
+import Link from "next/link";
+import Image from "next/image";
+import { Template } from "../../utils/getAllTemplates";
+import { StarUnfilled, StarFilled } from "../../assets/icons";
+
+interface Props {
+  template: Template;
+  isFavorite: boolean;
+  handleFavoriteTemplates: any;
+}
+
+export default function TemplateCard({
+  template,
+  isFavorite,
+  handleFavoriteTemplates,
+}: Props) {
+  return (
+    <div className="group relative h-60 w-full overflow-hidden pb-4 md:w-[48%] lg:h-72">
+      <div className="relative overflow-hidden rounded">
+        <Image
+          src={template.src}
+          layout="responsive"
+          width={1200}
+          height={636}
+          sizes="40vw"
+          quality={100}
+        />
+
+        <button
+          className="pointer-events-all absolute bottom-2 right-2 z-20 flex rounded-full bg-zinc-800/70 p-1 outline outline-2 outline-black transition-transform duration-100 active:scale-110"
+          onClick={() => handleFavoriteTemplates(template.name)}
+        >
+          {isFavorite ? StarFilled : StarUnfilled}
+        </button>
+      </div>
+
+      <p className="my-2 text-left text-sm font-medium capitalize lg:text-base">
+        {template.name}
+      </p>
+
+      <div className="my-4 flex justify-between gap-x-4 overflow-hidden p-px transition-[margin] duration-700 group-hover:mb-10">
+        <button className="template-action-button -translate-x-72 bg-zinc-50 text-zinc-800 md:translate-x-0 md:-translate-y-24">
```

Comment:
Not sure if these styles for buttons, paragraphs, links, etc repeat. But if you ever going to reuse it anywhere in the app, it should be a component.

You can get inspiration for components from MUI lib for example. Button, Typography, etc